Subject: [PATCH] Make PIN/password failed attempts dialog non-cancelable

This commit makes it harder for a users pocket to trigger an MDM
remote wipe, by setting the warning dialog to noncancelable.

With this change the pocket would first need to tap on OK before
being able to attempt to enter a PIN. This mirrors the existing
behaviour of both PIN and PUK warning dialogs.

Test: manual - enable an MDM set to trigger a wipe of the device/user
on unsucessful unlock attempts. Then incorrectly enter your PIN until
the warning dialog appears.  The dialog should only be dismissable
by pressing the 'OK' button.

diff --git a/packages/Keyguard/src/com/android/keyguard/KeyguardSecurityContainer.java b/packages/Keyguard/src/com/android/keyguard/KeyguardSecurityContainer.java
index a7e4e12..61e2fc0 100644
--- a/packages/Keyguard/src/com/android/keyguard/KeyguardSecurityContainer.java
+++ b/packages/Keyguard/src/com/android/keyguard/KeyguardSecurityContainer.java
@@ -178,6 +178,7 @@
         final AlertDialog dialog = new AlertDialog.Builder(mContext)
             .setTitle(title)
             .setMessage(message)
+            .setCancelable(false)
             .setNeutralButton(R.string.ok, null)
             .create();
         if (!(mContext instanceof Activity)) {
